vendor/src/github.com/docker/swarmkit/api/ca.proto

```diff
@@ -43,6 +43,9 @@ message IssueNodeCertificateRequest {
 	// Token represents a user-provided string that is necessary for new
 	// nodes to join the cluster
 	string token = 3;
+
+	// Availability allows a user to control the current scheduling status of a node.
+	string availability = 4;
```

Comment:
I don't like much `availability` leaking to the `NodeCA`.

Also - if we go down this path, we should probably take an initial NodeSpec rather than a bunch of fields